api/src/integrationTest/java/uk/gov/hmcts/reform/divorce/caseapi/controllers/SolicitorSubmitPetitionControllerTest.java

```diff
@@ -0,0 +1,142 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.util.ResourceUtils;
+import uk.gov.hmcts.reform.divorce.caseapi.clients.FeesAndPaymentsClient;
+import uk.gov.hmcts.reform.divorce.caseapi.config.WebMvcConfig;
+import uk.gov.hmcts.reform.divorce.caseapi.config.interceptors.RequestInterceptor;
+import uk.gov.hmcts.reform.divorce.caseapi.service.SolicitorSubmitPetitionService;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.reform.divorce.caseapi.TestConstants.AUTH_HEADER_VALUE;
+import static uk.gov.hmcts.reform.divorce.caseapi.TestConstants.SERVICE_AUTHORIZATION;
+import static uk.gov.hmcts.reform.divorce.caseapi.TestConstants.SUBMIT_PETITION_API_URL;
+import static uk.gov.hmcts.reform.divorce.caseapi.caseapi.util.TestDataHelper.callbackRequest;
+import static uk.gov.hmcts.reform.divorce.caseapi.caseapi.util.TestDataHelper.getFeeResponse;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureWireMock(port = 0)
+@AutoConfigureMockMvc
+public class SolicitorSubmitPetitionControllerTest {
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private SolicitorSubmitPetitionService solicitorSubmitPetitionService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private FeesAndPaymentsClient feesAndPaymentsClient;
+
+    @MockBean
+    private RequestInterceptor requestInterceptor;
+
+    @MockBean
+    private WebMvcConfig webMvcConfig;
+
+    @BeforeEach
+    public void setUp() {
+        WireMock.reset();
+    }
+
+    @Test
+    public void givenValidCaseDataWhenCallbackIsInvokedThenOrderSummaryIsSet() throws Exception {
+        stubFor(
+            get(
+                urlEqualTo(
+                    "/fees-register/fees/lookup"
+                        + "?channel=default&event=issue&jurisdiction1=family&jurisdiction2=family%20court&service=divorce"
```

Comment:
I thought our jurisdiction was divorce but I assume this is correct

Reply:
yeah fees and pay team have set it up like that.